patroni/dcs/kubernetes.py

```diff
@@ -1321,11 +1338,26 @@ def touch_member(self, data: Dict[str, Any]) -> bool:
 
         member = cluster and cluster.get_member(self._name, fallback_to_leader=False)
         pod_labels = member and member.data.pop('pod_labels', None)
+
+        replaced_xlog_location: Optional[str] = data.get('xlog_location', None)
```

Comment:
What happens if `replaced_xlod_location is None`? Do we never update in such a case?

Reply:
We do update, if `deep_compare` on line 1353 decides the whole new and old member data is not equal (and that includes the case when the old xlog location is set, but the new is None); however, we never cache the `None` value of `xlog_location`, as it might be updated next moment with the actual location.